xformers/helpers/hierarchical_configs.py

```diff
@@ -19,15 +19,16 @@ class BasicLayerConfig:
     stride: int
     padding: int
     seq_len: int
+    feedforward: str
 
 
 def get_hierarchical_configuration(
-    layer_basic_configs: List[BasicLayerConfig],
+    layer_base_configs: List[BasicLayerConfig],
     layernorm_style: LayerNormStyle = LayerNormStyle.Pre,
     use_rotary_embeddings: bool = True,
     mlp_multiplier: int = 4,
-    dim_head=32,
-    feedforward="MLP",
+    in_channels: int = 3,
```

Comment:
depending on whether there's a layer prior to the hierarchical config (this is the case for the efficientformers, though not in the small cifar10 examples, you can have a prior conv-pool), the number of channels in the inputs can change